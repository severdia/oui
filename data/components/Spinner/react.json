{"description":"Generates an `spinner` element with/without overlay and optinaly\nusing props for sizes small or tiny for smaller sizes.","displayName":"Spinner","methods":[],"props":{"hasOverlay":{"type":{"name":"bool"},"required":false,"description":"boolean class for establish overlay"},"size":{"type":{"name":"enum","value":[{"value":"'small'","computed":false},{"value":"'tiny'","computed":false}]},"required":false,"description":"various sizes"},"testSection":{"type":{"name":"string"},"required":false,"description":"Hook for automated JavaScript tests"}},"private":false}