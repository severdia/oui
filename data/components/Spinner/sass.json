{"name":"Spinner\n","description":"Loading spinner. Target `.#{$namespace}overlay` to position\nas needed.","start":{"line":1,"column":1},"end":{"line":36,"column":4},"example":[{"type":"html","description":"Default spinner","code":"<div class=\"position--relative height--100\">\n  <div class=\"#{$namespace}overlay\">\n    <div class=\"#{$namespace}spinner\"></div>\n  </div>\n </div>"},{"type":"html","description":"Small spinner","code":"<div class=\"position--relative height--100\">\n  <div class=\"#{$namespace}overlay\">\n    <div class=\"#{$namespace}spinner #{$namespace}spinner--small\"></div>\n  </div>\n </div>"},{"type":"html","description":"Tiny spinner","code":"<div class=\"position--relative height--100\">\n  <div class=\"#{$namespace}overlay\">\n    <div class=\"#{$namespace}spinner #{$namespace}spinner--tiny\"></div>\n  </div>\n </div>"},{"type":"html","description":"Spinner with overlay\nThis gets inserted into an element and blocks user interaction. Requires a\nparent `div` with positioning (`absolute` or `relative`).","code":"<div class=\"position--relative\">\n   <div>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Non cum iusto repudiandae earum porro reprehenderit perspiciatis iste delectus ipsam, accusantium ad sunt, rem mollitia, omnis illum explicabo facere quibusdam qui.</div>\n   <div class=\"#{$namespace}overlay\">\n     <div class=\"#{$namespace}spinner #{$namespace}spinner--small\"></div>\n   </div>\n</div>"}]}