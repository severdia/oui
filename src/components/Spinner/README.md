# Spinner

Use spinners to indicate the system is taking action, such as loading data or waiting for the server to respond. Spinners are the most common form of infinite progress indicators.