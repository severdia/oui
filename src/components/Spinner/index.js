import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

/**
 * Generates an `spinner` element with/without overlay and optinaly
 * using props for sizes small or tiny for smaller sizes.
 * @param {Object} props - Properties passed to component
 * @returns {ReactElement}
 */

const Spinner = ({
  size,
  hasOverlay,
  testSection,
}) => {
  const classes = classNames(
    'oui-spinner',
    {[`oui-spinner--${size}`]: size}
  );
  if (hasOverlay) {
    return (
      <div
        data-oui-component={ true }
        data-test-section={ testSection }
        className="oui-overlay">
        <div className={ classes }></div>
      </div>
    );
  }
  return <div data-test-section={ testSection } className={ classes }></div>;
};

Spinner.propTypes = {
  /** boolean class for establish overlay */
  hasOverlay: PropTypes.bool,
  /** various sizes */
  size: PropTypes.oneOf(['small', 'tiny']),
  /** Hook for automated JavaScript tests */
  testSection: PropTypes.string,
};

Spinner.displayName = 'Spinner';

export default Spinner;
