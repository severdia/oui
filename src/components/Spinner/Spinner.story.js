import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs, select } from '@storybook/addon-knobs';

import Spinner from './index.js';

const stories = storiesOf('Spinner', module);
stories
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <div id="root-preview">
      {story()}
    </div>
  ));

stories.add('Default', (() => {
  return <Spinner data-test-section="test-spinner" />;
}));

stories.add('Smaller sizes', () => {
  return <Spinner size={ select('size', { small: 'small', tiny: 'tiny' }, 'small') } />;
});

stories.add('With overlay', () => {
  return (
    <div className="position--relative">
      <div>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Non cum iusto repudiandae
      earum porro reprehenderit perspiciatis iste delectus ipsam, accusantium ad sunt,
      rem mollitia, omnis illum explicabo facere quibusdam qui.</div>
      <Spinner hasOverlay={ true } />
    </div>
  );
});
