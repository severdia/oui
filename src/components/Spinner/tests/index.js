import React from 'react';
import Spinner from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('components/Spinner', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Spinner />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render correctly with overlay', () => {
    const output = shallow(
      <Spinner hasOverlay={ true } />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render correctly with size property set to tiny', () => {
    const output = shallow(
      <Spinner size='tiny' />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render correctly with size & overlay props', () => {
    const output = shallow(
      <Spinner hasOverlay={ true } size='tiny' />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render correctly including testSection prop', () => {
    const output = shallow(
      <Spinner testSection='testing-foo' />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
