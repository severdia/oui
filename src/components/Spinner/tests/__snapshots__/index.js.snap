// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Spinner should render correctly 1`] = `
<div
  className="oui-spinner"
/>
`;

exports[`components/Spinner should render correctly including testSection prop 1`] = `
<div
  className="oui-spinner"
  data-test-section="testing-foo"
/>
`;

exports[`components/Spinner should render correctly with overlay 1`] = `
<div
  className="oui-overlay"
  data-oui-component={true}
>
  <div
    className="oui-spinner"
  />
</div>
`;

exports[`components/Spinner should render correctly with size & overlay props 1`] = `
<div
  className="oui-overlay"
  data-oui-component={true}
>
  <div
    className="oui-spinner oui-spinner--tiny"
  />
</div>
`;

exports[`components/Spinner should render correctly with size property set to tiny 1`] = `
<div
  className="oui-spinner oui-spinner--tiny"
/>
`;
